src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  color: string;
+  shape: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides are not valid (side <= 0)');
+    }
+
+    const sides = [this.a, this.b, this.c];
+
+    let longestSide: number = sides[0];
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 1; i < sides.length; i++) {
+      if (sides[i] > longestSide) {
+        longestSide = sides[i];
+      }
 
-export class Circle {
+      if (i === 2) {
+        sides.splice(sides.indexOf(longestSide), 1);
+      }
+    }
 
+    if (longestSide >= sides.reduce((
+      sum: number,
+      currentSide:number,
+    ) => sum + currentSide)) {
+      throw new Error(
+        'The longest side of a triangle is >= than a sum of two others',
+      );
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(semiPerimeter
+    * (semiPerimeter - this.a)
+  * (semiPerimeter - this.b)
+  * (semiPerimeter - this.c));
+
+    return +area.toFixed(2);
```

Comment:
use math.floor and implement heper function to calculate this, because the same operation repeats 3 times